src/core/simulation.jl

```diff
@@ -983,21 +1067,145 @@ function _execute!(sim::Simulation; kwargs...)
-                end
-                IS.@record :simulation_status SimulationStepEvent(
-                    get_current_time(sim),
-                    step,
-                    "done",
-                )
-            end
+                    ProgressMeter.update!(progress_bar, global_stage_execution_count;
+                        showvalues = [(:Step, step), (:Stage, stage_name),
+                    (:("Simulation Timestamp"), get_current_time(sim))])
+                end #execition stage timer
+            end # execution order for loop
+            IS.@record :simulation_status SimulationStepEvent(
+                get_current_time(sim),
+                step,
+                "done",
+            )
+        end # Execution step timer
+    end # Steps for loop
+    return nothing
+end
+
+function _initialize_stage_storage!(sim::Simulation, store, cache_size_mib)
+    sequence = sim.sequence
+    execution_order = sequence.execution_order
+    executions_by_stage = sequence.executions_by_stage
+    horizons = sequence.horizons
+    intervals = sequence.intervals
+    order = sequence.order
+
+    stages = OrderedDict{Symbol, SimulationStoreStageParams}()
+    stage_reqs = Dict{Symbol, SimulationStoreStageRequirements}()
+    stage_order = [order[x] for x in sort!(collect(keys(order)))]
+    num_param_containers = 0
+    rules = CacheFlushRules(max_size = cache_size_mib * MiB)
+    for stage_name in stage_order
+        num_executions = executions_by_stage[stage_name]
+        horizon = horizons[stage_name]
+        stage = sim.stages[stage_name]
+        psi_container = get_psi_container(stage)
+        duals = get_constraint_duals(psi_container.settings)
+        parameters = get_parameters(psi_container)
+        variables = get_variables(psi_container)
+        num_rows = num_executions * get_steps(sim)
+
+        interval = intervals[stage_name][1]
+        resolution = get_resolution(stage)
+        system = get_system(stage)
+        base_power = PSY.get_base_power(system)
+        sys_uuid = IS.get_uuid(system)
+        stage_params = SimulationStoreStageParams(
+            num_executions,
+            horizon,
+            interval,
+            resolution,
+            base_power,
+            sys_uuid,
+        )
+        reqs = SimulationStoreStageRequirements()
+
+        # TODO: configuration of keep_in_cache and priority are not correct
+        stage_sym = Symbol(stage_name)
+        for name in duals
+            array = get_constraint(psi_container, name)
+            reqs.duals[Symbol(name)] = _calc_dimensions(array, name, num_rows, horizon)
+            add_rule!(
+                rules,
+                stage_sym,
+                STORE_CONTAINER_DUALS,
+                name,
+                false,
+                CachePriority.LOW,
+            )
         end
-        sim_results = SimulationResultsReference(sim)
+
+        for (name, param_container) in parameters
```

Comment:
`parameters` here can be `nothing` which creates an error. Can we either:
1. add logic to skip this for loop if parameters is nothing
2. modify `get_parameters` to return an empty dict

Reply:
I addressed this one in 26018df